src/fidesops/service/connectors/saas_query_config.py

```diff
@@ -191,19 +270,28 @@ def generate_update_stmt(
         self, row: Row, policy: Policy, request: PrivacyRequest
     ) -> SaaSRequestParams:
         """
-        Prepares the update request by masking the fields in the row data based on the policy.
-        This masked row is then added as the body to a dynamically generated SaaS request.
+        This returns the method, path, header, query, and body params needed to make an API call.
+        The fields in the row are masked according the the policy and added to the request body
+        if specified by the body field of the masking request.
         """
 
-        current_request: SaaSRequest = self.masking_request
+        current_request: SaaSRequest = self.get_masking_request()
         collection_name: str = self.node.address.collection
-        collection_values: Dict[str, Row] = {collection_name: row}
+
+        # A combined dictionary of the collections that have already been retrieved
+        # along with the current row. This is to support the case where an update
+        # needs reference values from other collections, not just the values in
+        # the current row.
+        collection_values: Dict[str, Row] = {
+            **self.get_cached_results(request),
```

Comment:
In this use case, the additional reference value comes from the customer collection, which only has a single row. This would not work if the reference collection has more that one row. Would it be worth it to build out a more general solution as part of Fidesops core?

Reply:
This feels pretty error prone, discussed elsewhere

Reply:
Removed the need for this. For Stripe, there is enough information on the row to build the update request, we don't need to access the cache results.